commands/volumes/volume-status.go

```diff
@@ -49,11 +51,32 @@ func checkStatus(ctx transaction.TxnCtx) error {
 		}
 
 		// TODO: Check actual brick status when we get them running.
-		fakeStatus := &brick.Brickstatus{
-			Online: false,
-			Pid:    1234,
+
+		port := pmap.RegistrySearch(binfo.Path, pmap.GfPmapPortBrickserver)
+
+		brickDaemon, err := brick.NewGlusterfsd(binfo)
+		if err != nil {
+			return err
+		}
+
+		online := false
+
+		pid, err := daemon.ReadPidFromFile(brickDaemon.PidFile())
+		if err == nil {
+			// Check if process is running
+			_, err := daemon.GetProcess(pid)
+			if err == nil {
+				online = true
+			}
+		}
+
```

Comment:
Can you reset port to -1 (or 0) here if you've found the brick to be not online ? The port information in pmap registry might be stale.